src/world/FieldComputations.cpp

```diff
@@ -185,125 +94,113 @@ std::shared_ptr<Vector2> FieldComputations::lineIntersectionWithDefenceArea(cons
+                                                            const double robotRadius, LineSegment goalSide) {
+    // Discard the robot if it belong to the same team or if it has the given id.
+    if (robot->getId() == id && robot->getTeam() == (ourTeam ? Team::us : Team::them)) return {};
+
+    // Discard already the robot if it is not between the goal and point, or if the robot is standing on this point.
+    double lenToBot = (point - robot->getPos()).length();
+    bool isRobotItself = lenToBot <= robotRadius;
+    bool isInPotentialBlockingZone = ourGoal ? robot->getPos().x < point.x + robotRadius : robot->getPos().x > point.x - robotRadius;
+    if (isRobotItself || !isInPotentialBlockingZone) return {};
+
+    /* Check which part of the goal is blocked by this robot, by creating vectors from the point to the places that intersect with the robot and expanding these vectors to an
+     * infinite line that intersects with the infinite line expansion of the goal side. After which intersection point of these lines is found and a projection is applied between
+     * the intersection point and the goal side. Note that this projection is used, because the intersection point might not be on the goal side. In which case the point has to be
+     * mapped to the bottommost or topmost position of the goal side. You can compare this code with computing the 'shadow' on the goal caused by the robot if the point is
+     * considered a 'light source'. Note also that we do not have to check whether the lines go parallel, because the isRobitItself and isInPotentialBlockingZone check guarantees
+     * that the lines will eventually intersect with each other and by this same check we know that the lines intersect in the right direction, i.e. if the robot was not in between
+     * that goal and the point then the lines can also intersect with the infinite line expansion of the goal side but never causes a blockade 'shadow' on the goal side. */
+    double theta = asin(robotRadius / lenToBot);
+    double length = sqrt(lenToBot * lenToBot - robotRadius * robotRadius);
+    Vector2 lowerSideOfRobot = point + Vector2(length, 0).rotate((Vector2(robot->getPos()) - point).angle() - theta);
+    Vector2 upperSideOfRobot = point + Vector2(length, 0).rotate((Vector2(robot->getPos()) - point).angle() + theta);
+    Vector2 lowerMapToGoal = goalSide.project(util::twoLineIntersection(point, lowerSideOfRobot, goalSide.start, goalSide.end));
+    Vector2 upperMapToGoal = goalSide.project(util::twoLineIntersection(point, upperSideOfRobot, goalSide.start, goalSide.end));
+
+    LineSegment blockade = LineSegment(lowerMapToGoal, upperMapToGoal);
+    if (blockade.length() > NEGLIGIBLE_LENGTH) {
+        return blockade;
+    } else {
+        return {}; // Ignore blockades 'shadows' that are entirely below/above the goal side.
```

Comment:
Consider explicitly returning std::nullopt
```suggestion
        return std::nullopt; // Ignore blockades 'shadows' that are entirely below/above the goal side.
```